### PR TITLE
chore: Bump most of our deps

### PR DESCRIPTION
**Pull request by poljar**
The notable exceptions are:
1. Zeroize - Blocked on vodozemac
2. xshell - Requires code changes
3. image - Blocked on [qrcode-rust](https://github.com/kennytm/qrcode-rust)
4. rqrr - Same as image

### CODE REVIEW

File: crates/matrix-sdk-crypto/Cargo.toml [hunk 1]
```diff
@@ -25,35 +25,40 @@ docsrs = []
 testing = ["http"]
 
 [dependencies]
-aes = { version = "0.7.4", features = ["ctr"] }
-aes-gcm = "0.9.2"
-atomic = "0.5.0"
+aes = "0.8.1"
+ctr = "0.9.1"
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/Cargo.toml:29]: Can we maintain / fix alphabetical ordering?

File: crates/matrix-sdk-crypto/Cargo.toml [hunk 2]
```diff
@@ -65,13 +70,13 @@ rev = "4014e6959"
 features = ["client-api-c", "rand", "unstable-msc2676", "unstable-msc2677"]
 
 [dev-dependencies]
-futures = { version = "0.3.15", default-features = false, features = [
+futures = { version = "0.3.21", default-features = false, features = [
     "executor",
 ] }
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/Cargo.toml:75]: This wouldn't get longer than other lines when putting it back onto one.

File: crates/matrix-sdk-test-macros/Cargo.toml [hunk 3]
```diff
@@ -15,6 +15,6 @@ version = "0.1.0"
 proc-macro = true
 
 [dependencies]
-proc-macro2 = "1.0.29"
-quote = "1"
-syn = { version = "1.0.81", features = ["full",] }
\ No newline at end of file
+proc-macro2 = "1.0.37"
+quote = "1.0.18"
+syn = { version = "1.0.91", features = ["full",] }
```
Comment (jplatte) [inline crates/matrix-sdk-test-macros/Cargo.toml:20]:
```suggestion
syn = { version = "1.0.91", features = ["full"] }
```

**Review by jplatte — APPROVED** [review]
Looks good except minor style concerns 👍🏼

**Review by gnunicorn — APPROVED** [review]
I don't think it is really necessary to bump all patch-levels all the time (unless there is a bug in there that we triggered in an early-patch release), nor for minor releases (unless we are using a feature from that release-level of course) as `cargo` would resolve to them anyways. Quite the opposite if any other crate is pinned to a specific version for whatever reason, we might become unresolvable for them. But I guess ahead of a release is a good time to review and bump.

Two minor detail questions to resolve.

File: crates/matrix-sdk-crypto/src/file_encryption/attachments.rs [hunk 4]
```diff
@@ -229,15 +236,15 @@ impl<'a, R: Read + ?Sized + 'a> AttachmentEncryptor<'a, R> {
...
-        let iv = GenericArray::from_slice(&*iv);
-        let key = GenericArray::from_slice(&*key);
+        let encoded_iv = Base64::new((iv).to_vec());
+
+        let key_array = &key.into();
 
-        let aes = Aes256::new(key);
-        let aes = Aes256Ctr::from_block_cipher(aes, iv);
+        let aes = Aes256Ctr::new(key_array, &iv.into());
+        key.zeroize();
```
[the remark this answers is not in the document]
Comment (dkasak) [reply crates/matrix-sdk-crypto/src/file_encryption/attachments.rs:249]: There's no need for the IV (nonce) to be kept secret so there's no need to zeroize it.

**Review by dkasak — APPROVED** [review]
Looks good from my side. I introduced some faff attempting to remove the use of `GenericArray` so it'd be clearer no copies are done (for zeroization), but then decided against it because then we need to check for the lengths twice and I convinced myself that the initial way was correct.

File: crates/matrix-sdk-crypto/Cargo.toml [hunk 5]
```diff
@@ -25,35 +25,40 @@ docsrs = []
...
+http = { version = "0.2.6", optional = true }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies.vodozemac]
 git = "https://github.com/matrix-org/vodozemac"
 rev = "e09c93f2c8df9770793abeec57ed984d5e1f3834"
 
+[target.'cfg(target_arch = "wasm32")'.dependencies.getrandom]
+version = "0.2.6"
+features = ["js"]
+
```
[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-crypto/Cargo.toml]: Right, fixed by https://github.com/matrix-org/matrix-rust-sdk/pull/618/commits/d6f9747ce181502b668dcaac63ad1c7628e704e4.

**Pull request merged**